### PR TITLE
Feature Branch: Game Termination, Screens, Lobby Flexibility, and Cheating Features

### LINKED ISSUE #21: Place Win or Lose screen

### LINKED ISSUE #63: View players' scores
Players should receive points for which monster they killed (1-3). 

These should be added together and displayed on the left side of the game

### LINKED ISSUE #64: Implement cheating feature
Implementation of the specific cheating function

### LINKED ISSUE #73: Termination of the game by max Round implementation
The game should end as soon as round 14 has been reached, then the player who has collected the most trophies wins.

### LINKED ISSUE #74: Extension of the cheating feature
This expansion introduces a system that allows players to report others for cheating. If a player is found guilty of cheating after being reported, they will receive a penalty. Conversely, if a player makes a false accusation, they will receive a penalty. This mechanism completes the cheat feature by balancing the risks and responsibilities of reporting.

Here's how the cheat feature works: When cheat mode is activated, the player deals double damage to enemies. Once a player has realized this increased damage, the server marks them as a cheater for 4 rounds. During these 4 rounds, other players have the opportunity to report the suspected cheater. If no reports are made within the 4 rounds, the player is no longer considered a cheater. This temporary marking ensures that players are not permanently labeled as cheaters throughout the game.

### LINKED ISSUE #75: Main menu optimizing

### LINKED ISSUE #76: Make startGame button unclickable until a valid username is entered

### LINKED ISSUE #77: Make Lobby creation flexible

### LINKED ISSUE #78: resolving conflicts between branches and refactoring

### PR DESCRIPTION
**Pull request by Manuel1598**
In this feature branch, we have implemented several new features and improvements:

1. **Game Termination by Max Rounds or Tower HP:** The game now has the capability to terminate after a specified maximum number of rounds or by the minimum Tower HP.
2. **Win and Lose Screens:** Added screens to display the results of the game, either win or lose.
3. **Flexible Lobby Creation:** Enhanced the lobby creation process to be more flexible and adaptable to different requirements.
4. **Cheating Feature:** Introduced a cheating feature where the player can decide to be a cheater and do double damage to the monsters. You can activate and deactivate cheat mode by clicking the switch.
5. **View Player Score:** Players can now view their scores during the game.
6. **Extended Cheating Feature:** Improved and extended the initial cheating feature.
7. **Main Menu Optimization:** Optimized the main menu and added game rules for better user guidance.
8. **Start Game Button Enhancement:** Made the start game button unclickable until a valid username is entered.

These updates are designed to enhance the overall gaming experience by providing better game flow control, clearer results presentation, and more flexibility in lobby management.

Closes #21 
Closes #63 
Closes #64 
Closes #73 
Closes #74 
Closes #75 
Closes #76 
Closes #77 
Closes #78

### CODE REVIEW

File: app/src/main/res/layout/options.xml
Comment (Y1Egger) [inline app/src/main/res/layout/options.xml:1]: Wird dies überhaupt noch gebraucht? Option Button wurde ja entfernt.

**Review by RenaT0Pl — APPROVED** [review]
Von meiner Seite aus ist alles in Ordnung! 👍

**Review by Y1Egger — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/options.xml
Comment (Manuel1598) [reply app/src/main/res/layout/options.xml:1]: Timon wollte da etwas mit der Musik machen

Comment (Y1Egger) [reply app/src/main/res/layout/options.xml:1]: dann passt es eh

**Review by Y1Egger — APPROVED** [review]

**Pull request merged**